### DISCUSSION TITLE
Why can RaiseEvent() method called with a subtype of RoutedEventArgs that is never registered?

### DISCUSSION THREAD

**Discussion by martinrhan** [AvaloniaUI/Avalonia#13449] [Q&A]
When reading the source code, I see in [this file](https://github.com/AvaloniaUI/Avalonia/blob/master/src/Avalonia.Base/Input/GestureRecognizers/ScrollGestureRecognizer.cs):
` Target!.RaiseEvent(new ScrollGestureEventArgs(_gestureId, vector));`
However, I searched though the code base, and there is nothing like 
`
public static readonly RoutedEvent<ScrollGestureEventArgs> ScrollGestureEvent =
    RoutedEvent.Register<SomeType, ScrollGestureEventArgs>(
        nameof(ScrollGesture),
        RoutingStrategies.Bubble);
`
So how is it posible if the type of Args is not registered? In addition, how can I subsribe to events of ScrollGestureEventArgs?

**Comment by maxkatz6** [comment]
There is: https://github.com/AvaloniaUI/Avalonia/blob/master/src/Avalonia.Base/Input/Gestures.cs#L44-L46
